cpp/src/io/orc/timezone.cpp

```diff
@@ -398,9 +398,22 @@ timezone_table build_timezone_transition_table(std::string const &timezone_name)
     offsets.push_back(utcoff);
     if (!earliest_std_idx && !tzf.ttype[idx].isdst) { earliest_std_idx = ttimes.size() - 1; }
   }
-  if (!earliest_std_idx) { earliest_std_idx = 1; }
-  ttimes[0]  = ttimes[earliest_std_idx];
-  offsets[0] = offsets[earliest_std_idx];
+
+  if (tzf.timecnt() != 0) {
+    if (!earliest_std_idx) { earliest_std_idx = 1; }
+    ttimes[0]  = ttimes[earliest_std_idx];
+    offsets[0] = offsets[earliest_std_idx];
+  } else {
+    if (tzf.typecnt() == 0 || tzf.ttype[0].utcoff == 0) {
+      // No transitions, offset is zero; Table would be a no-op.
+      // Return an empty table to speed up parsing.
+      return {};
+    }
+    // No transitions to use for the time/offset - use the first offset and apply to all timestamps
+    CUDF_EXPECTS(tzf.typecnt() != 0, "Need at least one offset");
```

Comment:
This check seems unnecessary since the if-statement above will catch `== 0`.

Reply:
you're right. Not sure what I was thinking.